trustfall_stubgen/src/root.rs

```diff
@@ -400,3 +403,100 @@ fn make_vertex_file(
 
     vertex_file.top_level_items.push(vertex);
 }
+
+fn ensure_no_vertex_name_keyword_conflicts(
+    querying_schema: &Schema,
+    adapter: Arc<SchemaAdapter<'_>>,
+) {
+    let query = r#"
+{
+    VertexType {
+        name @output
+    }
+}"#;
+    let variables: BTreeMap<String, String> = Default::default();
+
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, serde::Deserialize)]
+    struct ResultRow {
+        name: String,
+    }
+
+    let mut rows: Vec<_> = trustfall::execute_query(querying_schema, adapter, query, variables)
+        .expect("invalid query")
+        .map(|x| x.try_into_struct::<ResultRow>().expect("invalid conversion"))
+        .collect();
+    rows.sort_unstable();
+
+    let mut uniq: HashMap<String, String> = HashMap::new();
+
+    for row in rows {
+        let name = row.name.clone();
+        let converted = escaped_rust_name(to_lower_snake_case(&name));
+        let v = uniq.insert(converted, name);
+        if let Some(v) = v {
+            panic!(
+                "cannot generate adapter for a schema containing both '{}' and '{}' vertices, consider renaming one of them",
+                v, &row.name
+            );
+        }
+    }
+}
+
+fn ensure_no_edge_name_keyword_conflicts(
```

Comment:
This function can probably use a better name, more in line with the panic message below